python/src/scipp/__init__.py

```diff
@@ -77,6 +77,18 @@ def custom_formatwarning(msg, *args, **kwargs):
 setattr(DataArray, 'sizes', property(_make_sizes))
 setattr(Dataset, 'sizes', property(_make_sizes))
 
+
+def _str_factory(cls):
+    def _str(obj):
+        return f'scipp.{cls.__name__}(sizes={obj.sizes})'
```

Comment:
'dim' instead of 'sizes'?

Also, keep it consistent with the HTML repr, don't mix braces and parens.

Reply:
On the Python side, `dims` refers only to the keys (dim labels).

Is it worth the extra code to replace the `{}` that Python uses for formatting dicts?